lesson18/index.js

```diff
@@ -88,6 +109,59 @@ const controlBtnBehavior = () => {
   nextBtn.classList.remove("is-disabled");
 }
 
+const setPagenation = (slides) => {
+  const slider = document.getElementById("js-slider");
+  const pagenation = document.createElement("div");
+  pagenation.classList.add("pagenation");
+  const pagenationWrapper = document.createElement("div");
+  pagenationWrapper.classList.add("pagenationWrapper");
+  const slideLength = slides.length;
+
+  slides.map((slide,index) => {
+    const span = document.createElement("span");
+    span.classList.add("pagenationBullet");
+    index === 0 ? span.classList.add("is-active") : span.classList.remove("is-active");
+    span.setAttribute("data-slideNo",`slideNo${index}`);
+    pagenationWrapper.appendChild(span);
+
+    span.addEventListener("click",(event)=>{
+      const bullets = document.querySelectorAll(".pagenationBullet");
+      const currentBullet = document.querySelector(".pagenationBullet.is-active");
+      const prevBtn = document.getElementById("js-slider-previousBtn");
+      const nextBtn = document.getElementById("js-slider-nextBtn");
+      const activateBulletIndex = Array.from(bullets).findIndex((bullet)=>{
+        return bullet === event.target;
+      })
+      currentBullet.classList.remove("is-active");
+      bullets[activateBulletIndex].classList.add("is-active");
+
+      if(activateBulletIndex === 0){
+        prevBtn.classList.add("is-disabled");
+        nextBtn.classList.remove("is-disabled");
+        return;
+      }
+      if(activateBulletIndex === slideLength){
+        nextBtn.classList.add("is-disabled");
+        prevBtn.classList.remove("is-disabled");
+        return;
+      }
+
+      controlSlideByIndex(index);
```

Comment:
IMO
The argument index is the same as activateBulletIndex, right?
If so, I would suggest passing activeBulletIndex instead of index. That way, there is no need for a map.